### PR TITLE
[Backup] `az backup container register`: Fix refresh container bug

### PR DESCRIPTION
**Pull request by akshayneema**
**Description**

- Fixed the refresh protectable container API call that happens when the protectable container is not found on the first try in az backup container register command. CRI link - [Incident 259315847 : [ARR] [SAP],[DISCOVERY], | [KRKEMA] | [Unable to discovery SAP Hana DB from Azure VM "xub29041431h0" using Azure CLI via ADO pipeline. Getting error message: Container unavailable or already registered.]](https://portal.microsofticm.com/imp/v3/incidents/details/259315847/home)

- Added 'MAB' in the allowed values for backup management type in az backup job/container list command

- Increased the sleep delay while tracking operations to prevent frequent calls.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
backup

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def load_arguments(self, _):
 
     with self.argument_context('backup container list') as c:
         c.argument('vault_name', vault_name_type, id_part=None)
-        c.argument('backup_management_type', backup_management_type)
+        c.argument('backup_management_type', arg_type=get_enum_type(allowed_backup_management_types + ["MAB"]), help=backup_management_type_help)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py:143]: Why not add `MAB` to `allowed_backup_management_types`?

**Comment by zhoxing-ms** [comment]
Could you add some tests for this change?

**Comment by zhoxing-ms** [comment]
Could you add the issue link in the description of this PR?

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 1 above)
Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py:143]: I'll check with the PM on this and get back.

**Comment by akshayneema** [comment]
> Could you add the issue link in the description of this PR?

Added CRI link in description.

**Comment by akshayneema** [comment]
> Could you add the issue link in the description of this PR?



> Could you add some tests for this change?

It's an internal call that we are making only for newly created SQL/HANA VMs. For testing, we use an already present setup for SQL and HANA cases so adding testing specifically for this won't be possible with the current testing setup we use.

**Comment by akshayneema** [comment]
> > Could you add some tests for this change?
> 
> It's an internal call that we are making only for newly created SQL/HANA VMs. For testing, we use an already present setup for SQL and HANA cases so adding tests specifically for this change won't be possible with the current testing setup we use.

In the future, if we move to an autogenerated setup it will get tested

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 1 above)
Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py:143]: We only support selected commands for MAB(Azure Backup Agent) backup management type. So, we've only added it in the  arg_type enum wherever necessary.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**